src/platform/datapath_raw_socket.c

```diff
@@ -392,7 +407,21 @@ CxPlatResolveRoute(
         "Starting to look up neighbor on Path[%hhu] with status %u",
         PathId,
         Status);
-    if (Status != ERROR_SUCCESS || IpnetRow.State <= NlnsIncomplete) {
+    //
+    // We need to force neighbor solicitation (NS) if any of the following is true:
+    // 1. No neighbor entry for the given destination address.
+    // 2. The neighbor entry isn't in usuable state.
+    // 3. In refresh scenario, the neighbor entry is the same as the existing one.
```

Comment:
The term "refresh" isn't well-established in the code. You've only used the word here and in a trace that says "Refreshed route has different local address", without defining the term anywhere.

Reply:
ok, not more "refreshed".